lib/aff4.py

```diff
@@ -392,6 +392,10 @@ def _UpdateChildIndex(self, urn, token):
 
         try:
           self.intermediate_cache.Get(urn)
+          # When an object is retrieved from the cache bump it back to the
+          # front to ensure heavily accessed items remain in the cache for their
+          # full age.
+          self.intermediate_cache.Put(urn, 1)
```

Comment:
This should not be necessary. The cache always moves all accessed items to the front of the list on access. Are you worried about the time expiring? That causes one write every 10 minutes in the current configuration, which should be ok, don't you think?